tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/LiferayGradleWorkspaceProject.java

```diff
@@ -49,13 +52,23 @@
 import org.eclipse.core.runtime.jobs.JobChangeAdapter;
 import org.eclipse.wst.server.core.ServerCore;
 
+import org.gradle.tooling.GradleConnectionException;
+import org.gradle.tooling.GradleConnector;
+import org.gradle.tooling.ProjectConnection;
+import org.gradle.tooling.model.DomainObjectSet;
+import org.gradle.tooling.model.GradleModuleVersion;
+import org.gradle.tooling.model.eclipse.EclipseExternalDependency;
+import org.gradle.tooling.model.eclipse.EclipseProject;
+
 /**
  * @author Andy Wu
  * @author Simon Jiang
  * @author Terry Jia
  */
 public class LiferayGradleWorkspaceProject extends LiferayWorkspaceProject {
 
+	public static List<Artifact> targetPlatformArtifacts = Collections.emptyList();
```

Comment:
Charles, can you work on a general 'projectProvider' caching strategy? this way we can cache calls to LiferayCore.create(IWorkspaceproject.class, project);

ANd so it won't create it over and over.  But we need to know when to 'evict' those cache entries.  So we can have a very efficient "listener", like this one in buildship
https://github.com/eclipse/buildship/blob/master/org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/workspace/SynchronizingBuildScriptUpdateListener.java

SO for each "project" the "provider" would tell the call, what resources (filters) it need sto list to in order to 'evict' the entry.  So each project would tell the "framework" what resources to listen for, so it will know when to 'evict' the proejct from the cache.

Can you work on this and use this for this implementation? 

Thanks.  WE can have a call tomorrow on this.